src/harness/testcases/heartbeat_testcase.py

```diff
@@ -273,6 +273,207 @@ def test_WINNF_FT_S_HBT_4(self):
+    self.assertEqual(response['cbsdId'], cbsd_id)
+    self.assertTrue(response['grantId'])
+    self.assertEqual(response['response']['responseCode'], 0)
+    self.assertTrue(response['measReportConfig'])
+    grant_id = response['grantId']
+    grant_expire_time = datetime.strptime(response['grantExpireTime'],
+                                          '%Y-%m-%dT%H:%M:%SZ')
+    del request, response
+
+    # Trigger to request measurement report for all subsequent heartbeat request
+    self._sas_admin.TriggerMeasurementReportHeartbeat({'meas_report_config':
+                                                       ['EUTRA_CARRIER_RSSI_ALWAYS']})
+    # First Heartbeat Request to Authorize Device
+    request = {
+        'heartbeatRequest': [{
+            'cbsdId': cbsd_id,
+            'grantId': grant_id,
+            'operationState': 'GRANTED'
+        }]
+    }
+    response = self._sas.Heartbeat(request)['heartbeatResponse'][0]
+
+    # Check the heartbeat response
+    self.assertTrue(response['measReportConfig'])
+    self.assertEqual(response['cbsdId'], cbsd_id)
+    self.assertEqual(response['grantId'], grant_id)
+    transmit_expire_time = datetime.strptime(
+      response['transmitExpireTime'], '%Y-%m-%dT%H:%M:%SZ')
+    self.assertLess(datetime.utcnow(), transmit_expire_time)
+    self.assertLessEqual(
+      (transmit_expire_time - datetime.utcnow()).total_seconds(), 240)
+    self.assertLessEqual(transmit_expire_time, grant_expire_time)
+    self.assertEqual(response['response']['responseCode'], 0)
+    del request, response
+
+    # Get measReport
+    meas_report = json.load(
+        open(os.path.join('testcases', 'testdata', 'meas_report_0.json')))
+
+    # Second Heartbeat Request with measReport
+    request = {
+      'heartbeatRequest': [{
+        'cbsdId': cbsd_id,
+        'grantId': grant_id,
+        'operationState': 'GRANTED',
+        'measReport': meas_report
+      }]
+    }
+    response = self._sas.Heartbeat(request)['heartbeatResponse'][0]
+
+    # Check the heartbeat response
+    self.assertTrue(response['measReportConfig'])
+    self.assertEqual(response['cbsdId'], cbsd_id)
+    self.assertEqual(response['grantId'], grant_id)
+    transmit_expire_time = datetime.strptime(response['transmitExpireTime'],
+                                             '%Y-%m-%dT%H:%M:%SZ')
+    self.assertLess(datetime.utcnow(), transmit_expire_time)
+    self.assertLessEqual(transmit_expire_time, grant_expire_time)
+    self.assertLessEqual(
+      (transmit_expire_time - datetime.utcnow()).total_seconds(), 240)
+    self.assertEqual(response['response']['responseCode'], 0)
+
+  @winnforum_testcase
+  def test_WINNF_FT_S_HBT_6(self):
+    """Array request: SAS-directed Heartbeat Req: SAS has requested CBSD to 
+    perform measurement through measReportConfig in Grant response or a 
+    previous Heartbeat Response.
+
+    The response should be SUCCESS.
+    """
+
+    # Register the device
+    device_a = json.load(
```

Comment:
Maybe device_e to match the file? Or device1 so that it's not implied that this is device_a.json?